c_src/quicer_nif.c

```diff
@@ -1047,12 +1098,62 @@ on_upgrade(ErlNifEnv *env,
 }
 
 /*
-** unload is called when the module code that the NIF library belongs to is
-*purged as old. New code of the same module may or may not exist.
-*/
+** on_unload is called when the module code that the NIF library belongs to is
+*  purged as old.
+*
+*  New code of the same module may or may not exist.
+*
+*  But there are three cases:
+*
+*  Case A: No new code of the same module exists.
+*          arg `priv_data` is not NULL.
+*
+*          It is ok to teardown the MsQuic with API handle and then close the
+*          API handle.
+*
+*  Case B: New code of the same module exists and it uses the same NIF DSO.
+*          arg `priv_data` is NULL.
+*
+*          It could be checked with `quicer:nif_mapped()`
+*
+*          It is *NOT* ok to teardown the MsQuic since the new code is
```

Comment:
it is actually *new*.  This is a tricky case I found while testing. 

When you load new beam file, it is not necessary come with new DSO, the new beam could ' link ' to the DSO, shared with old one. 
like 
```
old beam ----> DSO (one copy in VM process mem)
new beam ---/
```
So when unload the old beam, we shall not teardown the msquic since new beam is still 'link' to it.